### PR TITLE
A couple of modular JVM fixes

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.jboss.org/browse/WFLY-10937
https://issues.jboss.org/browse/WFLY-10811

### CODE REVIEW

**Comment by marekkopecky** [comment]
I started internal Scripts TS. In the meantime, I can do some sanity checks. But we can provide final update from our side after TS will be finished. Scripts TS run can takes ~4-5 hours.

**Comment by marekkopecky** [comment]
I see shell scripts issues related with common.sh. I see also some issues with ps1 scripts related with return value of CLI script. These issues may be related with [PR for wf-core scripts](https://github.com/wildfly/wildfly-core/pull/3488/). I will check these issues tomorrow.

**Comment by jamezp** [comment]
@marekkopecky Were you seeing like `
bin/standalone.sh: 3: bin/common.sh: Syntax error: "(" unexpected`? If so that got fixed in #3490 [WFCORE-4077](https://issues.jboss.org/browse/WFCORE-4077) and merged into core 6.0.1.Final.

**Review by jamezp — CHANGES_REQUESTED** [review]
I've tested the scripts on Fedora 28 with bash and dash on Java 8 and Java 11. I also tested both the batch and PowerShell scripts with Java 8 and Java 9. All seemed to work minus the one change required for the `appclient.bat`.

File: feature-pack/src/main/resources/content/bin/appclient.bat [hunk 1]
```diff
@@ -83,6 +83,12 @@ if "x%JAVA_HOME%" == "x" (
   set "JAVA=%JAVA_HOME%\bin\java"
 )
 
+rem set default modular jvm parameters
+setlocal EnableDelayedExpansion
+call "!DIRNAME!common.bat" :setDefaultModularJvmOptions "!JAVA_OPTS!"
+set "JAVA_OPTS=!JAVA_OPTS! !DEFAULT_MODULAR_JVM_OPTIONS!"
+setlocal DisableDelayedExpansion
+
```
[the remark this answers is not in the document]
Comment (ropalka) [reply feature-pack/src/main/resources/content/bin/appclient.bat]: Thank you @jamezp for identifying these issues - fixed.
I also removed commit that is addressed in https://github.com/wildfly/wildfly/pull/11617

**Review by jamezp — APPROVED** [review]

**Comment by marekkopecky** [comment]
I add logs to scripts:
```diff
diff --git a/appclient.sh b/appclient.sh
index 04ba96d..1d6b939 100755
--- a/appclient.sh
+++ b/appclient.sh
@@ -89,7 +89,7 @@ if [ "x$JAVA" = "x" ]; then
         JAVA="java"
     fi
 fi
-
+echo "Java opts = $JAVA_OPTS"
 # Set default modular JVM options
 setDefaultModularJvmOptions $JAVA_OPTS
 JAVA_OPTS="$JAVA_OPTS $DEFAULT_MODULAR_JVM_OPTIONS"
diff --git a/common.sh b/common.sh
index 957cf6f..288e338 100755
--- a/common.sh
+++ b/common.sh
@@ -2,6 +2,7 @@
 
 setDefaultModularJvmOptions() {
   $JAVA --add-modules=java.se -version > /dev/null 2>&1 && MODULAR_JDK=true || MODULAR_JDK=false
+  echo "JAVA opts from setDefaultModularJvmOptions: $1"
   if [ "$MODULAR_JDK" = "true" ]; then
     DEFAULT_MODULAR_JVM_OPTIONS=`echo $1 | $GREP "\-\-add\-modules"`
     if [ "x$DEFAULT_MODULAR_JVM_OPTIONS" = "x" ]; then
```

And I start some script:
```
$ ./appclient.sh 
Java opts = -Xms64m -Xmx512m -XX:MaxMetaspaceSize=256m -Djava.net.preferIPv4Stack=true -Djboss.modules.system.pkgs=org.jboss.byteman -Djava.awt.headless=true
JAVA opts from setDefaultModularJvmOptions: -Xms64m
```

If I add quotes, script works as expected:
```diff
diff --git a/appclient.sh b/appclient.sh
index 04ba96d..3a22f2e 100755
--- a/appclient.sh
+++ b/appclient.sh
@@ -89,9 +89,9 @@ if [ "x$JAVA" = "x" ]; then
         JAVA="java"
     fi
 fi
-
+echo "Java opts = $JAVA_OPTS"
 # Set default modular JVM options
-setDefaultModularJvmOptions $JAVA_OPTS
+setDefaultModularJvmOptions "$JAVA_OPTS"
 JAVA_OPTS="$JAVA_OPTS $DEFAULT_MODULAR_JVM_OPTIONS"
 
 # Check for -d32/-d64 in JAVA_OPTS
diff --git a/common.sh b/common.sh
index 957cf6f..288e338 100755
--- a/common.sh
+++ b/common.sh
@@ -2,6 +2,7 @@
 
 setDefaultModularJvmOptions() {
   $JAVA --add-modules=java.se -version > /dev/null 2>&1 && MODULAR_JDK=true || MODULAR_JDK=false
+  echo "JAVA opts from setDefaultModularJvmOptions: $1"
   if [ "$MODULAR_JDK" = "true" ]; then
     DEFAULT_MODULAR_JVM_OPTIONS=`echo $1 | $GREP "\-\-add\-modules"`
     if [ "x$DEFAULT_MODULAR_JVM_OPTIONS" = "x" ]; then
```
```
[mkopecky@dhcp-10-40-4-197 bin] (master *)$ ./appclient.sh 
Java opts = -Xms64m -Xmx512m -XX:MaxMetaspaceSize=256m -Djava.net.preferIPv4Stack=true -Djboss.modules.system.pkgs=org.jboss.byteman -Djava.awt.headless=true
JAVA opts from setDefaultModularJvmOptions: -Xms64m -Xmx512m -XX:MaxMetaspaceSize=256m -Djava.net.preferIPv4Stack=true -Djboss.modules.system.pkgs=org.jboss.byteman -Djava.awt.headless=true
```

So basically, all scripts needs to be fixed (add quotes), right?

**Comment by ropalka** [comment]
This is the fix PR for common.sh issues: https://github.com/wildfly/wildfly-core/pull/3492

**Comment by jamezp** [comment]
@marekkopecky So if I understand correctly we need the change @ropalka made to use `$*` to pass all parameters as well as quote the argument passed to the function in `common.sh`. Is that correct?

**Comment by ropalka** [comment]
We don't need to quote function arguments with fix provided in https://github.com/wildfly/wildfly-core/pull/3492 @jamezp

**Pull request merged**